test/attr_comparable_test.rb

```diff
@@ -97,4 +97,50 @@ def initialize(last_name, first_name)
       assert @d2 != @d3
     end
   end
+
+  describe "parameters that are incompatible to order" do
+    before do
+      @d1 = ComparableTestOneParameter.new(false)
+      @d2 = ComparableTestOneParameter.new(true)
+      @d3 = ComparableTestManyParameters.new(false, 'D')
+      @d4 = ComparableTestManyParameters.new(true, 'D')
+      @d5 = ComparableTestManyParameters.new('Kelly', false)
+      @d6 = ComparableTestManyParameters.new('Kelly', true)
+    end
```

Comment:
The `before` block is for common code. For the lines here that are different, it's better to put them in the tests that use them. Note that you can also make private methods that return what you want. That's nice because those can be mixed and matched in different test cases.